webapp/channels/src/components/announcement_bar/text_dismissable_bar.tsx

```diff
@@ -20,67 +20,57 @@ interface Props extends Partial<AnnouncementBarProps> {
     className?: string;
 }
 
-type State = {
-    dismissed: boolean;
-}
-
-export default class TextDismissableBar extends React.PureComponent<Props, State> {
-    constructor(props: Props) {
-        super(props);
-
-        this.state = {
-            dismissed: true,
-        };
-    }
+const TextDismissableBar = (props: Props) => {
+    const {allowDismissal, text, ...extraProps} = props;
+    const [dismissed, setDismissed] = useState<boolean>(false);
 
-    static getDerivedStateFromProps(props: Props) {
-        const dismissed = localStorage.getItem(localStoragePrefix + props.text?.toString());
-        return {
-            dismissed: (dismissed === 'true'),
-        };
-    }
+    useEffect(() => {
+        const dismissedFromStorage = localStorage.getItem(localStoragePrefix + text?.toString());
+        if (dismissedFromStorage === 'true') {
+            setDismissed(true);
+        }
+    }, [text]);
 
-    handleDismiss = () => {
-        if (!this.props.allowDismissal) {
+    const handleDismiss = () => {
+        if (!allowDismissal) {
             return;
         }
         trackEvent('signup', 'click_dismiss_bar');
 
-        localStorage.setItem(localStoragePrefix + this.props.text?.toString(), 'true');
-        this.setState({
-            dismissed: true,
-        });
-        if (this.props.onDismissal) {
-            this.props.onDismissal();
+        localStorage.setItem(localStoragePrefix + text?.toString(), 'true');
+        setDismissed(
+            true,
+        );
+        if (props.onDismissal) {
+            props.onDismissal();
         }
```

Comment:
`onDismissal` doesn't seem to be used by `AnnouncementBar`, so instead of getting it from props, expand it at the argument definition.